lib/ui/painting.dart

```diff
@@ -5774,6 +5774,18 @@ base class _NativeCanvas extends NativeFieldWrapperClass1 implements Canvas {
   @Native<Void Function(Pointer<Void>)>(symbol: 'Canvas::save', isLeaf: true)
   external void save();
 
+  static Rect _sorted(Rect rect) {
+    if (rect.isEmpty) {
```

Comment:
This will be true if we have a zero rect. I'm not sure how much it really matters to capture that case, but if we're optimizing we might as well avoid the native calls for the zero rect.

Reply:
We'd have to add a test for stroke vs fill. You still stroke a zero rect. This is approximately the same as SkCanvas where they just sort it and pass it along and then the filtering against zero happens when it figures out whether it wants to fill or stroke. Given the complication, I'm happy just having it sort the rect at this stage rather than testing here for both zero and negative independently and then factoring in the operation type.

Reply:
Hmmm, I suppose it isn't too complicated and would only happen for inverted rects in the first place. How expensive is the native call?

Reply:
The code changes were pretty minor so I went with the suggestion...